ecosystem/typescript/sdk/src/plugins/coin_client.ts

```diff
@@ -54,8 +61,32 @@ export class CoinClient {
       // If this is the first time an account has received the specified coinType,
       // and this is set to false, the transaction would fail.
       createReceiverIfMissing?: boolean;
+      // fungible asset address if you wish to transfer a fungible asset
+      assetAddress?: MaybeHexString;
```

Comment:
I wonder if we can merge this with coinType? We can check if coinType is just an address or containing the fully qualified name (address::module::struct)

Reply:
in the new fungible asset world, 
`coinType` is the asset type (default to 0x1::fungible_asset::Metadata)
`assetAddress` is the asset address

i.e https://github.com/aptos-labs/aptos-core/pull/8294/files#diff-cb86fcb259943ce56d237a9a0b938ce90ea148ab8a7e16f1545dff606c3fa659R37-R41
`assetAddress = assetAddress`
`coinTypes = assetType`

Reply:
I know that but coinType is a string here and thus it could be either 0x1::aptos_coin::AptosCoin or 0x123 (representing some fungible asset). Wouldn't that be easier to use and understand?

Reply:
sorry not following, what we would do then if someone passes coinType (i.e assetType) and assetAddress? we need to use both

Reply:
Why do we need to use both?
1. If user passes a coinType of the form 0x1::aptos_coin::AptosCoin, this means they want to transfer coin v1
2. If user passes a coinType that's an address (0x123), we can assume they want to transfer fungible assets

Reply:
and if they pass a coinType that's an address (0x123) and also want to set the asset type and not use the default asset type?

Reply:
What do you mean by setting the asset type? Isn't the address the fungible asset type?

Reply:
hmm no.. there is an assetType and assetAddress https://github.com/aptos-labs/aptos-core/pull/8294/files#diff-cb86fcb259943ce56d237a9a0b938ce90ea148ab8a7e16f1545dff606c3fa659R37-R40

assetAddress is an object address (0x123) and asset type is by [default](https://github.com/aptos-labs/aptos-core/pull/8294/files#diff-cb86fcb259943ce56d237a9a0b938ce90ea148ab8a7e16f1545dff606c3fa659R11) `0x1::fungible_asset::Metadata` but can be changed to a different type if needed

Reply:
assetType is not really required when sending a tx from off chain so defaulting to 0x1::fungible_asset::Metadata is okay. Passing a custom type is only really useful on chain in Move code

Reply:
looking at this [entry](https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/primary_fungible_store.move#L140-L144) function, we need to pass a type (`<T: key>`). SDK defaults to  `0x1::fungible_asset::Metadata` but gives the option to override it and pass any type they want.

Reply:
Hmm, in that case how about naming it as fungibleAssetType to be clear?

Reply:
_but IMO that's confusing and would not support the case when the user wants to provide both._

That is what I say. We give the user the option to provide 2 "things"
1. asset address - to transfer an amount of assets from
2. asset type - the asset type

We default it to a `0x1::fungible_asset::Metadata` type but the user can override it if they need to. Because of that, we have 2 arguments.

Reply:
I said you could give the user that option to override but `Metadata` always works so actually they are no benefit of letting them override. 
So if we don't let them override, what @movekevin suggests makes sense.

Reply:
Yeah I don't really see any benefits in letting users override when sending a tx or calling a view function. The ability to pass custom type arg is meant to be more useful when writing code in Move